test/functional/plugin/lsp_spec.lua

```diff
@@ -2205,20 +2220,24 @@ describe('LSP', function()
 
       helpers.mkdir_p(old_dir)
 
-      local file = "file"
+      local file = 'file'
       write_file(old_dir .. pathsep .. file, 'Test content')
 
-      exec_lua([[
+      local lines = exec_lua([[
         local old_dir = select(1, ...)
         local new_dir = select(2, ...)
+	local pathsep = select(3, ...)
+	local oldbunr = vim.fn.bufadd(old_dir .. pathsep .. 'file')
 
         vim.lsp.util.rename(old_dir, new_dir)
-      ]], old_dir, new_dir)
+	return vim.fn.bufloaded(oldbufnr)
+      ]], old_dir, new_dir, pathsep)
 
+      eq(0, lines)
       eq(false, exec_lua('return vim.loop.fs_stat(...) ~= nil', old_dir))
       eq(true, exec_lua('return vim.loop.fs_stat(...) ~= nil', new_dir))
       eq(true, exec_lua('return vim.loop.fs_stat(...) ~= nil', new_dir .. pathsep .. file))
-      eq('Test content', read_file(new_dir .. pathsep .. file))
+      eq('Test content\n', read_file(new_dir .. pathsep .. file))
```

Comment:
@justinmk I have merge a test, but now I started getting the new file content with a new line added. Not sure what cause of it. Maybe you have an idea?

Reply:
Adding an extension to a filename helped. Not sure why it didn't cause any errors before